src/redux/compare/state.ts

```diff
@@ -0,0 +1,143 @@
+import {K8sResource} from '@models/k8sresource';
+
+export const initialState: CompareState = {
+  isOpen: false,
+  views: [],
+  current: {
+    view: {
+      operation: 'union',
+      leftSet: undefined,
+      rightSet: undefined,
+    },
+    selection: [],
+    transfering: {
+      pending: false,
+    },
+  },
+};
+
+export type CompareState = {
+  isOpen: boolean;
+  views: SavedComparisonView[];
+  current: {
+    view: ComparisonView | SavedComparisonView;
+    left?: ResourceSetData;
+    right?: ResourceSetData;
+    comparison?: ComparisonData;
+    selection: string[];
+    inspect?: ComparisonInspection;
+    search?: string;
+    filtering?: {
+      comparisons: ResourceComparison[];
+    };
```

Comment:
I've been looking at https://redux-toolkit.js.org/api/createEntityAdapter and maybe we should consider adopting this pattern more.

Something along the lines of:
```
current: {
  comparisons: {
    ids: string[];
    filtered: string[];
    selected: string[];
    entities: ResourceComparison[];
  }
}
```